extensions/ql-vscode/src/queries-panel/query-tree-view-item.ts

```diff
@@ -3,24 +3,43 @@ import * as vscode from "vscode";
 export class QueryTreeViewItem extends vscode.TreeItem {
   constructor(
     name: string,
-    public readonly path: string,
-    language: string | undefined,
+    public readonly path: string | undefined,
     public readonly children: QueryTreeViewItem[],
   ) {
     super(name);
-    this.tooltip = path;
-    if (this.children.length === 0) {
-      this.description = language;
-      this.collapsibleState = vscode.TreeItemCollapsibleState.None;
-      this.contextValue = "queryFile";
-      this.command = {
-        title: "Open",
-        command: "vscode.open",
-        arguments: [vscode.Uri.file(path)],
-      };
-    } else {
-      this.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
-      this.contextValue = "queryFolder";
-    }
   }
 }
+
+export function createQueryTreeNodeItem(
```

Comment:
I think I would personally prefer using the domain names here, namely QueryTreeFolderItem and QueryTreeFileItem. Right now I have to translate 'node' and 'leaf' in my head to folder and file to have the correct image in my head. However, I don't have a CS background and could imagine it doesn't make sense to call it something other than node and leaf? For me folder and file is just more descriptive.

Reply:
I don't have a strong preference between the names. I was just picking one and my mind goes towards nodes and leaves because we're making `TreeView`/`QueryTree`.

I do now see we're using "queryFolder" and "queryFile" for the context values. I'll change the function names to match those.